app/assets/javascripts/features/machines.js

```diff
@@ -0,0 +1,49 @@
+const buttons = [...document.querySelectorAll('button')]
+const solutions = [...document.querySelectorAll('.problem__solutions')]
+const headers = [...document.querySelectorAll('.device__problems header')]
+
+var first_device = {
+  device: document.querySelector('.device__1'),
+  list: document.querySelector('.device-1__problems')
+}
+var second_device = {
+  device: document.querySelector('.device__2'),
+  list: document.querySelector('.device-2__problems')
+}
+var third_device = {
+  device: document.querySelector('.device__3'),
+  list: document.querySelector('.device-3__problems')
+}
+
+var devices = [first_device, second_device, third_device]
+
+const removeThis = i => {
+  devices.forEach(i => {
+    i.device.classList.remove('device-selected')
+    i.list.classList.add('__inactive')
+  })
+}
+
+const addThis = i => {
+  i.device.classList.add('device-selected')
+  i.list.classList.remove('__inactive')
+}
```

Comment:
`addThis` is also a very vague function name

Some of the same comments here: change `i` to `device` and use `device.button`, and change `-inactive`